org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/InvisibleProjectBuildSupport.java

```diff
@@ -43,15 +46,26 @@ public boolean fileChanged(IResource resource, CHANGE_TYPE changeType, IProgress
 			return false;
 		}
 		refresh(resource, changeType, monitor);
-		IProject invisibleProject = resource.getProject();
-		IPath realFolderPath = invisibleProject.getFolder(ProjectUtils.WORKSPACE_LINK).getLocation();
-		if (realFolderPath != null) {
-			IPath libFolderPath = realFolderPath.append(LIB_FOLDER);
-			if (libFolderPath.isPrefixOf(resource.getLocation())) {
-				UpdateClasspathJob.getInstance().updateClasspath(JavaCore.create(invisibleProject), libFolderPath, monitor);
+		String resourcePath = resource.getLocation().toOSString();
```

Comment:
It is inappropriate to use `toOSString()` here, because the glob used to match is `ProjectUtils.resolveGlobPath(base, pattern): IPath`, and `IPath.toString()` will always use separator `/`.

So here resourcePath should also use `toString()` to ensure separator is `/`. Btw, this is the only occurance of `toOSString()` in this PR.